docs/Versions.md

```diff
@@ -0,0 +1,15 @@
+# Versions
+Now that [Homebrew/versions](https://github.com/homebrew/homebrew-versions) has been deprecated [Homebrew/core](https://github.com/homebrew/homebrew-core) now supports multiple versions of formulae with a new naming format.
+
+In [Homebrew/versions](https://github.com/homebrew/homebrew-versions) the formula for GCC 6 was named `gcc6.rb` and began `class Gcc6 < Formula`. In [Homebrew/core](https://github.com/homebrew/homebrew-core) this same formula is named `gcc@6.rb` and begins `class GccAT6 < Formula`.
+
+## Acceptable Versioned Formulae
+Homebrew's versions are not intended to be used for any old versions you personally require for your project; formulae submitted should be expected to be used by a large number of people and still supported by their upstream projects.
+
+You should create your own [tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md) for formulae you or your organisation wishes to control the versioning of or those that do not meet the above standards.
+
+Versioned formulae we include must meet the following standards:
+
+* Versioned formulae should differ in major/minor (not patch) versions from the current stable release. This is because patch versions indicate bug or security updates and we do not wish to allow you to avoid applying security updates.
+* Formulae that depend on versioned formulae must not depend on the same formulae at two different versions twice in their recursive dependencies. For example, if you depend on `openssl@1.0` and `foo` and `foo` depends on `openssl` then you must instead use `openssl`.
```

Comment:
Can the following aspects of the standards be clarified? (I know @MikeMcQuaid will be familiar with this but added details/links for other readers) I wrote a couple situations and used months to clarify what might happen over time:

1. Can the version/head status of `openssl` as compared to `openssl@1.0` be clarified here? 
2. When will versions simply be updated, and when will foo go from simply `foo` to `foo@1.0`,  `foo@2.0` and `foo`? 
3. Also, what if in january `foo` and `bar` depends on `openssl@1.0`, while `fizz` depends on `foo` and `bar`, and then in february `openssl@2.0` is released, in march `foo` supports `openssl@2.0` but and `bar` will not support `openssl@2.0` until november.  What happens to `fizz` in april? 
4. What if a new `buzz` pull request is made in April with the same dependencies as `fizz`?
5. Can a new very popular `bop` repository in the exact same situation as `buzz`, add a new version for `foo`, `bar`, and `openssl`, which all require something like `openssl@1.5` in May?
6. What about item 5 in December after `bar` is updated to `openssl@2.0`?

The above situations happen quite often in ROS and in the robotics/scientific domains in general for libraries like OpenCV, VTK, PCL, and Gazebo, many of which may depend on libraries like Boost, Qt, Eigen, ceres-solver (which depends on strict Eigen versions), and the various BLAS implementations. Each of the above also have many other libraries built on top of subsets of the real libraries listed so far. You can verify this in the issues section of: https://github.com/mikepurvis/ros-install-osx